devito/dse/rewriters.py

```diff
@@ -406,6 +411,36 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        Skew the accesses along the time Dimension.
+        """
+
+        skew_factor = 3  # TOFIX
+
+        if skew_factor not in range(0, self.MAX_SKEW_FACTOR):
+            raise ValueError("skew_factor must be less than %d" % self.MAX_SKEW_FACTOR)
+
+        _, sub_iterators, directions = cluster.ispace.args
+
+        skew_dim, mapper, intervals = None, {}, []
+
+        for i in cluster.ispace.intervals:
+            if i.dim.is_Time:
+                intervals.append(Interval(i.dim, 0, 0))
+                skew_dim = i.dim
+            else:
+                mapper[i.dim] = i.dim + skew_factor*skew_dim
+                intervals.append(Interval(i.dim, -skew_factor*skew_dim,
+                                          -skew_factor*skew_dim))
+
+        ispace = IterationSpace(intervals, sub_iterators, directions)
+        cluster = Cluster(cluster.exprs, ispace, cluster.dspace, cluster.atomics,
+                          cluster.guards)
+        processed = xreplace_indices(cluster.exprs, mapper)
+        return cluster.rebuild(processed)
```

Comment:
This way (including the lines above) still doesn't make much sense, conceptually